docs/command_reference.md

```diff
@@ -0,0 +1,299 @@
+```
+
+The `--role` and `--gun` flags can be provided to specify a role and GUN to import the key to, which overrides any information contained in PEM headers.
+Note that for root and delegation keys, the `--gun` flag is ignored because these keys can be shared across GUNs.
+If no `--role` or `--gun` is given, notary will assume that the key is to be used for a delegation role.
+
+
+Moreover, it's possible for notary to import multiple keys contained in one PEM file, each separated into separate PEM blocks.
+
+For each key it attempts to import, notary will prompt for a passphrase so that the key can be encrypted at rest.
+
+Notary can also export all of its encrypted keys, or individually by key ID or GUN:
+```bash
+# export all my notary keys to a file
+$ notary key export -o exported_keys.pem
+
+# export a single key by ID
+$ notary key export --key <keyID> -o exported_keys.pem
+
+# export multiple keys by ID
+$ notary key export --key <keyID1> --key <keyID2> -o exported_keys.pem
+
+# export all keys for a GUN
+$ notary key export --gun <GUN> -o exported_keys.pem
+
+# export all keys for multiple GUNs
+$ notary key export --gun <GUN1> --gun <GUN2> -o exported_keys.pem
+```
+When exporting multiple keys, all keys are outputted to a single PEM file in individual blocks. If the output flag `-o` is omitted, the PEM blocks are outputted to STDOUT.
+
+## Manage keys for delegation roles
+
+To delegate content signing to other users without sharing the targets key, retrieve a x509 certificate for that user and run:
+
+```bash
+$ notary delegation add -p <GUN> targets/<role> user.pem user2.pem --all-paths
+```
+
+The delegated user can then import the private key for that certificate keypair (using `notary key import`) and use it for signing.
+
+The `--all-paths` flag allows the delegation role to sign content into any target name.  To restrict this, you can provide path prefixes with the `--paths` flag instead.  For example:
+```bash
+$ notary delegation add -p <GUN> targets/<role> user.pem user2.pem --paths tmp/ --paths users/
+```
+In the above example, the delegation would be allowed to sign targets prefixed by `tmp/` and `users/` (ex: `tmp/file`, `users/file`, but not `file`)
+
+It's possible to add multiple certificates at once for a role:
+```bash
+$ notary delegation add -p <GUN> targets/<role> --all-paths user1.pem user2.pem user3.pem
+```
+
+You can also remove keys from a delegation role, such that those keys can no longer sign targets into the delegation role:
+
+```bash
+# Remove the given keys from a delegation role
+$ notary delegation remove -p <GUN> targets/<role> <keyID1> <keyID2>
+
+# Alternatively, you can remove keys from all delegation roles
+$ notary delegation purge <GUN> --key <keyID1> --key <keyID2>
+```
+
+## Managing targets in delegation roles
+
+We can specify to sign content into specific delegation roles by specifying the `--roles` flag.  This also applies to `notary addhash` and `notary remove`.
```

Comment:
nitpick: maybe "we can specify which delegation roles to sign content into by using the `--roles` flag"?